spec/facility_spec.rb

```diff
@@ -2,25 +2,276 @@
+
+      end
+
+      it 'only renew if written and license are true in registrant data' do
+        expect(@registrant_1.license_data[:renewed]).to eq false
+        expect(@facility_1.renew_drivers_license(@registrant_1)).to eq false
+
+        @facility_1.administer_written_test(@registrant_1)
+        @facility_1.administer_road_test(@registrant_1)
+        expect(@registrant_1.license_data[:written]).to eq true
+        expect(@registrant_1.license_data[:license]).to eq true
+
+        expect(@facility_1.renew_drivers_license(@registrant_1)).to eq false
+        @facility_1.add_service('Renew License')
+        expect(@facility_1.renew_drivers_license(@registrant_1)).to eq true        
+      end
+
+      it 'updates the registrants license info if they renew' do
+        @facility_1.add_service('Renew License')
+        @facility_1.administer_written_test(@registrant_1)
+        @facility_1.administer_road_test(@registrant_1)
+        @facility_1.renew_drivers_license(@registrant_1)
+        expect(@registrant_1.license_data[:renewed]).to eq true
+      end
+    end
+  end
+
+  describe 'analysing registration data' do
+    before(:each) do
+      @vehicle_collection = DmvDataService.new.wa_ev_registrations
+      @factory = VehicleFactory.new
+      @factory.create_vehicles(@vehicle_collection)
+      @vehicles_to_be_registered = @factory.vehicle_lot
+      @facility_1.add_service('Vehicle Registration')
+      @vehicles_to_be_registered.each do |vehicle|
+        @facility_1.register_vehicle(vehicle)
+      end
+      
+    end
+
+    it 'registeres all vehicles from data set' do
+      expect(@facility_1.registered_vehicles.count).to eq 1000
+      expect(@facility_1.registered_vehicles). to include Vehicle
+    end
+
+    it 'can generate an array of all values matching a specific attribute' do
+      expect(@facility_1.all_of_one_attribute('year')).to be_an_instance_of Array
+      expect(@facility_1.all_of_one_attribute('year').count).to eq 1000
+    end
+
+    it 'can show list of unique values per key' do
+      # require 'pry';binding.pry
+      expect(@facility_1.unique_types('year')).to be_an_instance_of Array
+      expect(@facility_1.unique_types('year').count).to eq 14
+      p @facility_1.unique_types('year').sort
+    end
+
+    it 'can reveal how many instances of a specific value there are' do
+      expect(@facility_1.quantity_unique_types('year')).to be_an_instance_of Hash
+      expect(@facility_1.quantity_unique_types('year').count).to eq 14
+      p @facility_1.quantity_unique_types('year')
+    end
+
+    it 'can reveal other information for vehicle registration' do
+      expect(@facility_1.quantity_unique_types('model')).to be_an_instance_of Hash
+      expect(@facility_1.quantity_unique_types('model').count).to eq 67
+      expect(@facility_1.quantity_unique_types('make')).to be_an_instance_of Hash
+      expect(@facility_1.quantity_unique_types('make').count).to eq 27
     end
```

Comment:
Great job adding tests to your additional methods you created. 
Remember to try to test explicitly for the return value you're expecting.